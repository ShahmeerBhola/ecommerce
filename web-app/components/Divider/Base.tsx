import React from 'react';

export type Props = {
  color?: 'bg-gray-80' | 'bg-red-100' | 'bg-gray-500';
  borderColor?: 'border-gray-80' | 'border-red-100' | 'border-gray-500';
  width?: 'w-20' | 'w-full';
  my?: 'my-0' | 'my-2' | 'my-4' | 'my-5' | 'my-6' | 'my-10';
  mx?: 'mx-0' | 'mx-auto';
  borderThickness?: 'border-1' | 'border-2';
  className?: string;
};

const BaseDivider: React.FunctionComponent<Props> = ({
  color = 'bg-red-100',
  borderColor = 'border-red-100',
  width = 'w-20',
  my = 'my-10',
  mx = 'mx-auto',
  borderThickness = 'border-2',
  className,
}) => {
  const classNames = `${my} ${width}${mx ? ` ${mx}` : ''} ${className}`;
  return <hr className={`${classNames} ${color} ${borderThickness} border-solid ${borderColor}`} />;
};

export default BaseDivider;
