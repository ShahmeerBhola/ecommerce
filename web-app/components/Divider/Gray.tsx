import React from 'react';
import BaseDivider, { Props as BaseDividerProps } from './Base';

export type Props = Omit<BaseDividerProps, 'color' | 'borderColor'>;

const GrayDivider: React.FunctionComponent<Props> = (props) => (
  <BaseDivider color="bg-gray-80" borderColor="border-gray-80" width="w-full" borderThickness="border-1" {...props} />
);

export default GrayDivider;
