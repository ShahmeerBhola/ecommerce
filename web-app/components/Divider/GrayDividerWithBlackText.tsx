import React from 'react';
import DividerWithText, { Props } from './GrayDividerWithText';

const GrayDividerWithBlackText: React.FunctionComponent<Props> = ({ text, options, children }) => (
  <DividerWithText
    text={text}
    options={{
      text: {
        color: 'text-black',
        weight: 'font-bold',
        size: 'text-lg',
        className: 'uppercase whitespace-no-wrap mr-10',
      },
      divider: options?.divider || {},
    }}
  >
    {children}
  </DividerWithText>
);

export default GrayDividerWithBlackText;
