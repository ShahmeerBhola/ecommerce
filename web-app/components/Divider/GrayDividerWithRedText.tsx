import React from 'react';
import DividerWithText, { Props } from './GrayDividerWithText';

const GrayDividerWithRedText: React.FunctionComponent<Props> = ({ text, options, children }) => (
  <DividerWithText
    text={text}
    options={{
      text: {
        color: 'text-red-100',
        weight: 'font-medium',
        className: 'whitespace-no-wrap mr-10',
      },
      divider: options?.divider || {},
    }}
  >
    {children}
  </DividerWithText>
);

export default GrayDividerWithRedText;
