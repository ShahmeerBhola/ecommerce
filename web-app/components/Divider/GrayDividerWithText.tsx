import React from 'react';
import Text from '../Text';
import { Props as TextProps } from '../Text';
import GrayDivider, { Props as GrayDividerProps } from './Gray';

export type Props = {
  text: string;
  options?: {
    text?: TextProps;
    divider?: GrayDividerProps;
  };
};

const GrayDividerWithText: React.FunctionComponent<Props> = ({ text, children, options }) => {
  return (
    <div className="flex w-full items-center">
      <Text {...(options?.text || {})}>{text}</Text>
      <GrayDivider {...(options?.divider || {})} />
      {children}
    </div>
  );
};

export default GrayDividerWithText;
