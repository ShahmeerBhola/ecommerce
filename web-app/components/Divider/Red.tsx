import React from 'react';
import BaseDivider, { Props } from './Base';

const RedDivider: React.FunctionComponent<Omit<Props, 'color' | 'borderColor'>> = (props) => (
  <BaseDivider color="bg-red-100" borderColor="border-red-100" {...props} />
);

export default RedDivider;
