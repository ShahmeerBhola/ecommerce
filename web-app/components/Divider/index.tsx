import GrayDivider from './Gray';
import GrayDividerWithBlackText from './GrayDividerWithBlackText';
import GrayDividerWithRedText from './GrayDividerWithRedText';
import RedDivider from './Red';

export { GrayDivider, GrayDividerWithBlackText, GrayDividerWithRedText, RedDivider };
