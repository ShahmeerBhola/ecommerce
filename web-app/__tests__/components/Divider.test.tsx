import React from 'react';
import renderer from 'react-test-renderer';
import Divider from '../../components/Divider';

describe('Divider component', () => {
  test('it renders properly', () => {
    const tree = renderer.create(<Divider />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly with non-default props', () => {
    const tree = renderer.create(<Divider color="gray-400" width="full" my="25" borderThickness="1" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
