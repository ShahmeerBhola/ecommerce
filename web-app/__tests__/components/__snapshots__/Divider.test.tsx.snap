// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider component it renders properly 1`] = `
<hr
  className="my-10 w-20 mx-auto border-2 border-solid border-red-100"
/>
`;

exports[`Divider component it renders properly with non-default props 1`] = `
<hr
  className="my-25 w-full mx-auto border-1 border-solid border-gray-400"
/>
`;
